### PR TITLE
fix: make DC hidden columns invisible initially

### PR DESCRIPTION
**Pull request by juancruz-f7l**
## Description

Fix column visibility in OneDataCollection table where columns marked with `hidden: true` by developers were incorrectly starting visible instead of hidden on first load. User preferences now properly override developer defaults while still respecting initial hidden states.

Screenshot 2026-01-14 at 11 27 32
Screenshot 2026-01-14 at 11 27 45

### Root Cause
Default table settings were initialized with `{ order: [], hidden: [] }`, which are truthy values. This prevented the fallback to developer-defined `hidden: true` properties on columns, causing all columns to appear visible on first load.

### Changes Made

**1. Updated default settings structure:**
- Changed from `{ order: [], hidden: [] }` to `{}` in `collectionViewRegistry.tsx`
- Made properties optional: `{ order?: ColId[], hidden?: ColId[] }`

**2. Fixed storage loading:**
- Changed check from `if (featureValue)` to `if (featureValue !== undefined)` 
- Ensures empty arrays `[]` are properly restored from localStorage

**3. Enhanced new column handling:**
- Added `getMergedHidden()` function that merges user preferences with developer defaults
- New columns with `hidden: true` remain hidden even when user has saved preferences
- Uses `settings.order` array to determine which columns existed when user saved preferences

**4. Fixed column positioning:**
- New columns now appear at the end of the table (not beginning)
- Uses `-1` detection from `indexOf()` to properly sort new columns

**5. Improved settings merge:**
- `TableSettings` now uses updater function with spread operator
- Preserves other settings when updating order/hidden

### Behavior

**Before:**
- Columns with `hidden: true` → Always visible on first load ❌
- User shows column → Preference saved but reverted on next load ❌

**After:**
- Columns with `hidden: true` → Start hidden (developer default) ✅
- User shows column → Stays visible on reload (user preference) ✅
- Developer adds new hidden column → Respects `hidden: true` even with existing user prefs ✅
- Reset button → Clears preferences, restores developer defaults ✅

### Test Coverage
Added 9 comprehensive tests covering:
- Developer defaults on first load
- User preferences (empty array vs specific columns)
- Reset to defaults
- New columns with/without `hidden: true`
- Empty array edge cases for both `order` and `hidden`

All 34 table column tests passing ✅

### CODE REVIEW

**Review by sergiocarracedo — APPROVED** [review]

File: packages/react/src/experimental/OneDataCollection/visualizations/collection/Table/hooks/__tests__/useColumns.settings.test.ts [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        settings,
+        allowSorting,
+        allowHiding
+      )
+    )
+
+    // Should respect user-defined order
+    expect(result.current.colsOrder).toEqual(["column3", "column1", "column2"])
+  })
+})
```
[the remark this answers is not in the document]
Comment (juancruz-f7l) [reply packages/react/src/experimental/OneDataCollection/visualizations/collection/Table/hooks/__tests__/useColumns.settings.test.ts]: Created test for that 👍

File: packages/react/src/experimental/OneDataCollection/__stories__/visualizations/table/table.stories.tsx [hunk 2]
```diff
@@ -99,11 +99,36 @@ export const TableWithNestedRecords: Story = {
...
+        allowColumnHiding: true,
+      },
+    })
+    return (
+      <ExampleComponent
+        tableAllowColumnHiding
+        visualizations={[mockVisualizations.table]}
+        id="table-column-hidden/v1"
+      />
+    )
```
[the remark this answers is not in the document]
Comment (juancruz-f7l) [reply packages/react/src/experimental/OneDataCollection/__stories__/visualizations/table/table.stories.tsx]: Brought frozen cols back.
